src/domain/context/context/dto/context.dto.create.ts

```diff
@@ -36,10 +37,13 @@ export class CreateContextInput {
     nullable: true,
     description: 'Set of References for the new Context.',
   })
+  @ValidateNested({ each: true })
+  @Type(() => CreateReferenceInput)
   references?: CreateReferenceInput[];
 
-  @Field({ nullable: true })
+  @Field(() => CreateLocationInput, { nullable: true })
   @IsOptional()
   @ValidateNested()
+  @Type(() => CreateLocationInput)
   location?: CreateLocationInput;
```

Comment:
Location is mandatory in Context entity and again optional in IContext - please review.